Added test test_ovs_restart_pcs_disable_enable.

alexgrom9393 opened the pull request:
Restarting ovs-agents with pcs disable/enable on controllers.

Steps:
1. Update default security group
2. Create router01, create networks net01: net01__subnet,
    192.168.1.0/24, net02: net02__subnet, 192.168.2.0/24 and
    attach them to router01.
3. Launch vm1 in net01 network and vm2 in net02 network
    on different computes
4. Go to vm1 console and send pings to vm2
5. Disable ovs-agents on all controllers, restart service
    neutron-plugin-openvswitch-agent on all computes, and enable
    them back. To do this, launch the script against master node
6. Check that pings between vm1 and vm2 are not interrupted or
    not more than 2 packets are lost

Commit:
Added test test_ovs_restart_pcs_disable_enable.

Restarting ovs-agents with pcs disable/enable on controllers.

Steps:
1. Update default security group
2. Create router01, create networks net01: net01__subnet,
    192.168.1.0/24, net02: net02__subnet, 192.168.2.0/24 and
    attach them to router01.
3. Launch vm1 in net01 network and vm2 in net02 network
    on different computes
4. Go to vm1 console and send pings to vm2
5. Disable ovs-agents on all controllers, restart service
    neutron-plugin-openvswitch-agent on all computes, and enable
    them back. To do this, launch the script against master node
6. Check that pings between vm1 and vm2 are not interrupted or
    not more than 2 packets are lost

gdyuldin commented on mos_tests/neutron/python_tests/test_ovs_restart.py:
```diff
@@ -0,0 +1,208 @@
+                'on vm {}.'.format(vm.name)
+
+            with open('/tmp/ovs_restart_ping.log', 'w') as log_file:
+                for line in res['stdout']:
```
`log_file.writelines(res['stdout]')`

gdyuldin commented on mos_tests/neutron/python_tests/test_ovs_restart.py:
```diff
@@ -0,0 +1,208 @@
+                'Can\'t get log file /tmp/ovs_restart_ping.log ' \
+                'on vm {}.'.format(vm.name)
+
+            with open('/tmp/ovs_restart_ping.log', 'w') as log_file:
```
let's parametrize file path (add argument to to context manager)

ababi4 replied:
In disable_ovs_agents_on_controllers method the 'pcs resource disable p_neutron-plugin-openvswitch-agent' operation, which goes before 'pcs resource enable p_neutron-plugin-openvswitch-agent', is absent.

gdyuldin commented on mos_tests/neutron/python_tests/test_ovs_restart.py:
```diff
@@ -0,0 +1,208 @@
+                sent_count, recieved_count = packets_data.groups()
+                break
+
+        assert packets_data is not None, (
```
This will break if no one line in log above.
Will be better to use for...else loop (end call pytest.fail in else)

Commit:
Fixes for test_ovs_restart_pcs_disable_enable.

Commit:
Modified scenario for test.

In test_ovs_restart_pcs_disable_enable ovs_agents should be disabled only
on one controller.

Commit:
Changed test scenatio

Restart openvswitch-agents with pcs disable/enable on controllers

Steps:
1. Update default security group
2. Create router01, create networks net01: net01__subnet,
192.168.1.0/24, net02: net02__subnet, 192.168.2.0/24 and
attach them to router01.
3. Launch vm1 in net01 network and vm2 in net02 network
on different computes
4. Go to vm1 console and send pings to vm2
5. Disable ovs-agents on all controllers, restart service
neutron-plugin-openvswitch-agent on all computes, and enable
them back. To do this, launch the script against master node.
6. Wait 30 seconds, send pings from vm1 to vm2 and check that
it is successful.

gdyuldin commented on mos_tests/neutron/python_tests/test_ovs_restart.py:
```diff
@@ -0,0 +1,161 @@
+    def disable_ovs_agents_on_controller(self):
+        """Disable openvswitch-agents on all controllers."""
+        controller = self.env.get_nodes_by_role('controller')[0]
+        assert controller is not None, 'No controllers have been found.'
```
If node is None it will fail before this check.

gdyuldin commented on mos_tests/neutron/python_tests/test_ovs_restart.py:
```diff
@@ -0,0 +1,161 @@
+        controller = self.env.get_nodes_by_role('controller')[0]
+        assert controller is not None, 'No controllers have been found.'
+
+        with self.env.get_ssh_to_node(controller.data['ip']) as remote:
```
You can use 
`with controller.ssh() as remote:`

gdyuldin commented on mos_tests/neutron/python_tests/test_ovs_restart.py:
```diff
@@ -0,0 +1,161 @@
+    def restart_ovs_agents_on_computes(self):
+        """Restart openvswitch-agents on all computes."""
+        computes = self.env.get_nodes_by_role('compute')
+        assert len(computes) > 0, 'No computes have been found.'
```
Let's make this check in mark or test class ot method
`@pytest.mark.check_env_('smth')`

gdyuldin commented on mos_tests/neutron/python_tests/test_ovs_restart.py:
```diff
@@ -0,0 +1,161 @@
+        for node in controllers:
+            with self.env.get_ssh_to_node(node.data['ip']) as remote:
+                result = remote.execute(
+                    '. openrc && pcs resource enable '
```
This command will enable agent on all computes. You don't need to execute it on each one.

Commit:
Applying comments for test.

ababi4 commented on mos_tests/neutron/python_tests/test_ovs_restart.py:
```diff
@@ -0,0 +1,158 @@
+                assert result['exit_code'] == 0
+
+    def enable_ovs_agents_on_controllers(self):
+        """Disable openvswitch-agents on all controllers."""
```
Please, change the comment to """Enable etc. """

Commit:
"Applying comments. Changed env checks."

For OvsBase is checked that one ore more computes exist in the env as it
can be required to deploy env with 1 compute for similar tests.
For TestOVSRestartTwoVms is checked that two or more computes exist in the env
as we should create 2 vms on 2 computes.
